core/domain/exp_services_test.py

```diff
@@ -1673,6 +1682,17 @@ class ExplorationCommitLogUnitTests(ExplorationServicesUnitTests):
         'post_commit_status': 'public'
     }
 
+    COMMIT_UPDATE_FIRST_PUBLISHED_MSEC_EXP_2 = {
```

Comment:
How about this: I have a separate function so that a new commit is added to the log only for current published explorations in which their first_contribution_msec is updated. For all incremental publication cases, the update of first_published_msec is within a publish event.